aspnetcore/performance/performance-best-practices.md

```diff
@@ -42,7 +42,22 @@ A common performance problem in ASP.NET Core apps is blocking calls that could b
 
 A profiler, such as [PerfView](https://github.com/Microsoft/perfview), can be used to find threads frequently added to the [Thread Pool](/windows/desktop/procthread/thread-pools). The `Microsoft-Windows-DotNETRuntime/ThreadPoolWorkerThread/Start` event indicates a thread added to the thread pool. <!--  For more information, see [async guidance docs](TBD-Link_To_Davifowl_Doc)  -->
 
-## Return IEnumerable\<T> or IAsyncEnumerable\<T>
+## Return large collections
+
+A webpage shouldn't load an arbitrary number of rows. When returning a collection of objects, consider whether it could lead to performance issues. Determine if the design could produce the following poor outcomes:
```

Comment:
This is nit-picky, so feel free to ignore, but the data isn't necessarily arranged in rows. I might rephrase this first sentence as "A webpage shouldn't load large amounts of data all at once," or something like that.

Reply:
```suggestion
A webpage shouldn't load large amounts of data all at once. When returning a collection of objects, consider whether it could lead to performance issues. Determine if the design could produce the following poor outcomes:
```